docs/debug.md

```diff
@@ -0,0 +1,48 @@
+# CMake Debugging
+
+Starting with CMake 3.27, debugging CMake is now supported in CMake Tools!
+
+The following documentation will help you understand the various ways you can debug CMake.
```

Comment:
General comment for this document -- consider rewording "debug CMake" to "debug CMake scripts and cache generation" or similar to avoid confusion with debugging actual the CMake executable, which is also technically possible but requires a C++ debugger.

Reply:
Changing.